evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -35,6 +35,7 @@ def __init__(self,
                  ensembling=False,
                  text_in_ensembling=False,
                  pipeline_params=None,
+                 custom_hyperparameters=None,
```

Comment:
Do we still need the distinction between `pipeline_params` and `_frozen_pipeline_parameters`? I think it's adding some unnecessary complexity. From what I can tell, `pipeline_params` is used for passing the ts pipeline params (which are already included in `_frozen_pipeline_parameters`) and for overriding the parameters of the first batch.  Can we use only one of these? I think this is what's causing a bug in `test_automl_respects_pipeline_parameters_with_duplicate_components` (see comment below).

Reply:
So I was actually in the process of consolidating this as part of #2166 due to `problem_configuration` using time series parameters as well as the pipeline instances, since I agree that it would make more sense to have those parameters passed via one argument!